src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -1694,13 +1680,26 @@ impl MooncakeTable {
         let table_auto_incr_ids = self.next_file_id..(self.next_file_id + new_file_ids_to_create);
         self.next_file_id += new_file_ids_to_create;
         let table_event_id = snapshot_payload.uuid;
+
+        // Record index merge event initiation.
+        if let Some(event_replay_tx) = &self.event_replay_tx {
```

Comment:
The best way to assert is to add a `TaskGuard`, which asserts all replay events are issued from the same tokio task
```rs
pub struct TaskGuard {
    /// ID for the current tokio task.
    task_id: Mutex<Option<tokio::task::Id>>,
}

impl TaskGuard {
    pub fn new() -> Self {
        Self {
            task_id: Mutex::new(None),
        }
    }

    /// Assert all invocations are from the same tokio task.
    pub async fn check(&self) {
        let cur_id = tokio::task::id();
        let mut guard = self.task_id.lock().await;
        if let Some(id) = *guard {
            assert_eq!(id, cur_id, "called from multiple Tokio tasks!");
        } else {
            *guard = Some(cur_id);
        }
    }
}
```